compiler/damlc/daml-opts/daml-opts-types/DA/Daml/Options/Types.hs

```diff
@@ -58,14 +59,17 @@ data Options = Options
     -- ^ Controls whether the scenario service server runs all checks
     -- or only a subset of them. This is mostly used to run additional
     -- checks on CI while keeping the IDE fast.
-  , optHlintEnabled :: Bool
-  -- ^ Whether or not to enable hlint
-  , optHlintDataDir :: Maybe FilePath
-    -- ^ Where hlint's base configuration file (hlint.yaml) resides
+  , optHlintUsage :: Maybe HlintUsage
```

Comment:
Why is this a `Maybe`? What is the difference between `Nothing` and `HlintDisabled`?

Reply:
In truth not much:
  - `--hlint-enabled f` => `Just HlintEnabled f`
  - `--hlint-disabled`  => `Just HlintDisabled`
  - `{- nothing -}`     => `Nothing`
The code for parsing the options looks like this:
```haskell
hlintEnabledOpt :: Parser HlintUsage
hlintEnabledOpt = HlintEnabled <$> strOption
  ( long "with-hlint"
    <> metavar "DIR"
    <> help "Enable hlint with hlint.yaml directory"
  )

hlintDisabledOpt :: Parser HlintUsage
hlintDisabledOpt = flag' HlintDisabled
  ( long "without-hlint"
    <> help "Disable hlint"
  )

hlintUsageOpt :: Parser (Maybe HlintUsage)
hlintUsageOpt = optional (hlintEnabledOpt <|> hlintDisabledOpt)
```
The sad truth is that I haven't yet been able to work out how to express an "optional with a default".

Reply:
```haskell
hlintEnabledOpt = option reader (long "with-hlint" <> metavar "DIR" <> help "Enable hlint with hlint.yaml directory" <> value HLintDisabled)
  where reader = HLintEnabled readerAsk
```
That should get you pretty close

Reply:
Another alternative would be to parse into a `Maybe` first and then convert that to `HLintUsage` using an `fmap` or just use a `Maybe` directly instead of `HLintUsage` (don’t feel particularly strongly about that either way).